### PR TITLE
Show more meanignful stack trace for ReactNative errors

### PR DESCRIPTION
**Pull request by bvaughn**
(_This is an upstream sync from a change committed to fbsource._)

Clicking on the stack should jump to where the error actually occurred rather than to where it's logged in a redbox component.

### Before
image

### After
image

### CODE REVIEW

File: src/renderers/shared/fiber/ReactFiberErrorLogger.js [hunk 1]
```diff
@@ -85,12 +93,10 @@ function logCapturedError(capturedError: CapturedError): void {
       `React caught an error thrown by one of your components.\n\n${error.stack}`,
     );
   }
-
-  showDialog(capturedError);
 }
 
 exports.injection = {
-  injectDialog(fn: (e: CapturedError) => void) {
+  injectDialog(fn: (e: CapturedError) => ?boolean) {
```
Comment (gaearon) [inline src/renderers/shared/fiber/ReactFiberErrorLogger.js]: Maybe let's make it strict?

Comment (bvaughn) [inline src/renderers/shared/fiber/ReactFiberErrorLogger.js]: Sure!

**Comment by bvaughn** [comment]
Going ahead and merging based on Tim's approval of this same change-set in fbsource.

**Comment by fakkiu** [comment]
💯 ship.it

**Pull request merged**